src/systems.rs

```diff
@@ -231,6 +233,39 @@ pub fn process_input_system(
         }
     }
 
+    fn push_ime_event(params: &mut ContextSystemParams, window: &Entity, event: egui::Event) {
+        params
+            .contexts
+            .get_mut(*window)
+            .unwrap()
+            .egui_input
+            .events
+            .push(event);
+    }
+
+    for ev in input_events.ev_ime_input.iter() {
+        match ev {
+            Ime::Preedit {
+                window,
+                value: _,
```

Comment:
shouldn't we also push a `CompositionUpdate` event with this value right after pushing `CompositionStart`?

(looking at the reference implementation at https://github.com/emilk/egui/blob/2bc2fb9c393bbbfc4832a35af1c100bc4ea48719/crates/egui-winit/src/lib.rs#L301)

Reply:
Sounds good to me. I read through the reference and all code that uses the event. The event is used by TextEdit to preview text before it's committed by the IME. It should also be added for the sake of completion.

Reply:
Sorry for forgetting to comment on this again, I feel really bad for neglecting this PR for several releases. The last question I want to ask is if there is a reason why we don't align our implementation to the reference one completely.

I haven't had a chance to test it myself and to understand if there's really any difference in behaviour, but I guess I'd feel safer if we just adopted exactly the same logic.